include/flatbuffers/flatc.h

```diff
@@ -33,7 +33,8 @@ class FlatCompiler {
   struct Generator {
     typedef bool (*GenerateFn)(const flatbuffers::Parser &parser,
                                const std::string &path,
-                               const std::string &file_name);
+                               const std::string &file_name,
+                               const flatbuffers::IDLOptions& opts);
```

Comment:
There's already a lot of generator related options in the IDLOptions in Parser, so yes, that is normal.